BrainPortal/lib/resource_access.rb

```diff
@@ -88,16 +82,19 @@ def find_accessible_by_user(id, user, options = {})
     # [For *admin* users:] any resource on the system.
     # [For regular users:] all resources that belong to a group to which the user belongs.
     def find_all_accessible_by_user(user, options = {})
-      scope = self.where(options) # will fail if not simple attibute mappings
+      scope = self.where(options) # will fail if not simple attribute mappings
+
+      return scope if user.has_role? :admin_user
 
-      unless user.has_role? :admin_user
-        scope = scope.joins(:user)
+      scope     = scope.joins(:user)
+      available_group_ids = (user.group_ids + Group.where(:public => true).pluck(:id)).uniq
```

Comment:
Should be replaced by one of the new methods ? seems to be `viewable_groups`?

Reply:
I don't know. This is tricky. I'll check later.